extmod/modtls_mbedtls.c

```diff
@@ -502,6 +524,7 @@ STATIC mp_obj_t ssl_socket_make_new(mp_obj_ssl_context_t *ssl_context, mp_obj_t
         mbedtls_ssl_free(&o->ssl);
         mp_raise_ValueError(MP_ERROR_TEXT("CERT_REQUIRED requires server_hostname"));
     }
+    mbedtls_ssl_set_verify(&o->ssl, &ssl_sock_cert_verify, o);
```

Comment:
Why not use `mbedtls_ssl_conf_verify()` and call it in `ssl_context_make_new()`?